File: desert/hoarding-gold.lua
function pos2xy(pos) return pos['x'], pos['y'] end

local totalGold = 0
while totalGold < 25 do
    local coin = self:findNearest(self:findItems())
    totalGold = totalGold + coin.value
    self:moveXY(pos2xy(coin.pos))
end

self:moveXY(38, 34)
self:say("Done collecting gold!")
self:moveXY(59, 33)
self:say(totalGold)

File: desert/hoarding-gold.clj
(def totalGold 0)
(while (< totalGold 25)
  (def coin (.findNearest this (.findItems this)))
  (if coin
    (when true
      (def totalGold (+ totalGold (.value coin)))
      (.moveXY this (.x (.pos coin)) (.y (.pos coin))))
    (.wait this 0.1)))
(.moveXY this 38 34)
(.say this "Done collecting gold!")
(.moveXY this 59 33)
(.say this totalGold)
